### PR TITLE
[#17] [Android] [UI] As a user, I can see the Side Menu screen after tapping on User Avatar in Home screen

### LINKED ISSUE #17: [Android] [UI] As a user, I can see the Side Menu screen after tapping on User Avatar in Home screen
## Why

The user should see the Side Menu screen after they tap on User Avatar in their Home screen

## Acceptance Criteria

Design: https://www.figma.com/file/GjRPOjDyZ6f4EDL3wKarRK/Challenge-Mobile-App?node-id=34%3A1579

### PR DESCRIPTION
**Pull request by luongvo**
- Close #17

## What happened 👀

- Add menu Drawer that wraps AppNavigation https://proandroiddev.com/navigation-drawer-using-jetpack-compose-27ea7db74903.
- Bind user info (name & avatar) on Side Menu.

## Insight 📝

- The original [ModelDrawer](https://foso.github.io/Jetpack-Compose-Playground/material/modaldrawer/) does not support opening from the right side. We need to use a workaround by using `CompositionLocalProvider(LocalLayoutDirection provides LayoutDirection.Rtl)` to force it reverse the swipe direction. I finally made a custom `RtlModalDrawer` https://gist.github.com/luongvo/f5cc7e679c0388996ea460969fb4335f.

## Proof Of Work 📹

https://user-images.githubusercontent.com/16315358/210269294-ff0ac898-6c77-4db7-ba2c-ed9d0f86affa.mp4

### CODE REVIEW

File: android/src/main/java/vn/luongvo/kmm/survey/android/ui/screens/MainScreen.kt [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+            style = typography.body1.copy(fontSize = 20.sp),
+            textAlign = TextAlign.Start,
+            modifier = Modifier
+                .fillMaxWidth()
+                .clickable { onLogoutClick() }
+                .padding(vertical = dimensions.paddingSmall)
+        )
+        Spacer(modifier = Modifier.weight(1f))
+        Text(
+            text = "v${BuildConfig.VERSION_NAME} (${BuildConfig.VERSION_CODE})",
```
Comment (manh-t) [inline android/src/main/java/vn/luongvo/kmm/survey/android/ui/screens/MainScreen.kt:112]: @luongvo Can we get the version from VM and put it here instead? 🙏

Comment (luongvo) [reply android/src/main/java/vn/luongvo/kmm/survey/android/ui/screens/MainScreen.kt:112]: @manh-t makes sense 👍

Comment (luongvo) [reply android/src/main/java/vn/luongvo/kmm/survey/android/ui/screens/MainScreen.kt:112]: @manh-t The update requires ViewModel integration, I fixed it in https://github.com/luongvo/kmm-survey/pull/72/commits/1ac9d0eb1aeb32b0a27101d97aa0dad59fd36f7c instead, please check it out 🙏

**Review by manh-t — APPROVED** [review]

File: android/src/main/java/vn/luongvo/kmm/survey/android/ui/screens/MainScreen.kt [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+private fun Drawer(
+    user: UserUiModel?,
+    onLogoutClick: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    Column(
+        modifier
+            .statusBarsPadding()
+            .fillMaxSize()
+            .background(Nero90)
```
Comment (suho) [inline android/src/main/java/vn/luongvo/kmm/survey/android/ui/screens/MainScreen.kt]: While we have dimensions and shapes, should we have the same for color?

Comment (luongvo) [reply android/src/main/java/vn/luongvo/kmm/survey/android/ui/screens/MainScreen.kt]: @suho Yes, do you mean this?

https://github.com/luongvo/kmm-survey/blob/e9197e77d45767f5073d18bf2bf76b0578d41368/android/src/main/java/vn/luongvo/kmm/survey/android/ui/theme/Color.kt#L9

Comment (luongvo) [reply android/src/main/java/vn/luongvo/kmm/survey/android/ui/screens/MainScreen.kt]: Hmm, I think we can move this generic color into AppColors with a specific name...

Comment (luongvo) [reply android/src/main/java/vn/luongvo/kmm/survey/android/ui/screens/MainScreen.kt]: Fixed in https://github.com/luongvo/kmm-survey/pull/70/commits/42604c09c7b17cadb8da2a8334aac3a462ef643e

**Review by blyscuit — APPROVED** [review]

File: android/src/main/java/vn/luongvo/kmm/survey/android/ui/screens/MainScreen.kt (hunk 2 above)
Comment (suho) [reply android/src/main/java/vn/luongvo/kmm/survey/android/ui/screens/MainScreen.kt]: Nice approach!

**Review by suho — APPROVED** [review]

**Pull request merged**